x-pack/plugins/apm/public/components/fleet_integration/apm_agents/runtime_attachment/java_agent_version_input.tsx

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import React, { useEffect } from 'react';
+import { i18n } from '@kbn/i18n';
+import { FormattedMessage } from '@kbn/i18n-react';
+import { EuiFlexItem, EuiFormRow, EuiLink, EuiComboBox } from '@elastic/eui';
+import { useKibana } from '@kbn/kibana-react-plugin/public';
+import { CoreStart } from '@kbn/core/public';
+import { createCallApmApi } from '../../../../services/rest/create_call_apm_api';
+import { FETCH_STATUS, useFetcher } from '../../../../hooks/use_fetcher';
+import { RuntimeAttachmentSettings } from '.';
+
+interface Props {
+  isValid: boolean;
+  version: string | null;
+  onChange: (nextVersion: RuntimeAttachmentSettings['version']) => void;
+}
+
+export function JavaAgentVersionInput({ isValid, version, onChange }: Props) {
+  const { services } = useKibana();
+
+  useEffect(() => {
+    createCallApmApi(services as CoreStart);
+  }, [services]);
+
+  const { data, status } = useFetcher((callApmApi) => {
+    return callApmApi('GET /internal/apm/fleet/java_agent_versions');
+  }, []);
+
+  useEffect(() => {
+    // When version is not available on the fleet package sets it to "latest"
+    if (version === null) {
+      // This is necessary due to a possible bug in Fleet where even thought the form is valid
+      // the save button is still disabled: https://github.com/elastic/kibana/issues/135131
+      setTimeout(() => {
+        onChange('latest');
+      }, 1);
+    }
+  }, [version, onChange]);
+
+  const isLoading = status === FETCH_STATUS.LOADING;
+  const options =
```

Comment:
nit `options` is such an overloaded term. What about `versions` or `agentVersions`:

```suggestion
  const agentVersions =
```